src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -213,8 +213,9 @@ private ConstantValue DefaultSyntaxValue
 
                     if (binder is not null && parameterEqualsValue is not null && !_lazyDefaultSyntaxValue.IsBad)
                     {
-                        NullableWalker.AnalyzeIfNeeded(binder, parameterEqualsValue, diagnostics);
-                        VerifyParamDefaultValueMatchesAttributeIfAny(_lazyDefaultSyntaxValue, parameterEqualsValue.Value.Syntax, diagnostics);
+                        var valueSyntax = parameterEqualsValue.Value.Syntax;
+                        NullableWalker.AnalyzeIfNeeded(binder, parameterEqualsValue, valueSyntax, diagnostics);
```

Comment:
>valueSyntax [](start = 85, length = 11)

Pulling syntax from a bound node? Please assert what syntax node we are dealing with here.